tests/e2e/utils/cognito_bootstrap/cognito_pre_deployment.py

```diff
@@ -97,31 +97,37 @@ def create_cognito_userpool(
 
 # Step 3: Configure Ingress
 def configure_ingress(cognito_userpool: CustomDomainCognitoUserPool, tls_cert_arn: str):
-
-    # annotate the ingress with ALB listener rule parameters
-    configure_env_file(
-        env_file_path="../../awsconfigs/common/istio-ingress/overlays/cognito/params.env",
-        env_dict={
+    ingress_values_file = "../../charts/common/ingress/cognito/values.yaml"
+    cognito_dict = {
             "CognitoUserPoolArn": cognito_userpool.arn,
             "CognitoAppClientId": cognito_userpool.client_id,
             "CognitoUserPoolDomain": cognito_userpool.userpool_domain,
             "certArn": tls_cert_arn,
-        },
+    }
+
+    # annotate the ingress with ALB listener rule parameters
+    configure_env_file(
+        env_file_path="../../awsconfigs/common/istio-ingress/overlays/cognito/params.env",
+        env_dict = cognito_dict
     )
 
+    write_env_to_yaml(cognito_dict, ingress_values_file)
 
 def configure_aws_authservice(
     cognito_userpool: CustomDomainCognitoUserPool, subdomain_name: str
 ):
+    aws_auth_service_values_file = "../../charts/common/aws-authservice/cognito/values.yaml"
+    logout_url_dict = {
+        "LOGOUT_URL": f"https://{cognito_userpool.userpool_domain}/logout?client_id={cognito_userpool.client_id}&logout_uri=https://kubeflow.{subdomain_name}"
+    }
     # substitute the LOGOUT_URL for the AWS AuthService to redirect to
     configure_env_file(
         env_file_path="../../awsconfigs/common/aws-authservice/base/params.env",
-        env_dict={
-            "LOGOUT_URL": f"https://{cognito_userpool.userpool_domain}/logout?client_id={cognito_userpool.client_id}&logout_uri=https://kubeflow.{subdomain_name}",
-        },
+        env_dict = logout_url_dict
     )
 
-
+    write_env_to_yaml(logout_url_dict, aws_auth_service_values_file)
```

Comment:
This will write to the kustomize params.env file and the the helm values.yaml file at the same time

Can we add a todo for decoupling the configuration logic?

Reply:
added comment in relevant places
```
#TO DO: The current script fills in Helm values and Kustomize params.env files at the same time. Need to decouple the two in future.  
```